api/genericresource/parse.go

```diff
@@ -0,0 +1,35 @@
+package genericresource
+
+import (
+	"strconv"
+	"strings"
+
+	"github.com/docker/swarmkit/api"
+)
+
+// Parse parses the GenericResource resources given by the arguments
+func Parse(tprs string, resources *api.Resources) error {
+	resources.Generic = make(map[string]*api.GenericResource)
+	mtpr := resources.Generic
+
+	for _, term := range strings.Split(tprs, ";") {
```

Comment:
I'm not sure splitting it in multiple flags would be the solution here as it would just bloat the code in swarmctl.

Shouldn't the user just say:
> --generic-resource "banana=blue,red,green;apple=8;orange=3f;"
or
> --generic-resource "banana=blue,red,green;apple=8;orange=5.67"

That would limit this to one flag and lift the ambiguity :)